drivers/storage/portworx/components_test.go

```diff
@@ -2677,43 +2679,90 @@ func TestSecurityInstall(t *testing.T) {
 		Spec: corev1alpha1.StorageClusterSpec{
 			Security: &corev1alpha1.SecuritySpec{
 				Enabled: true,
+				Auth: &corev1alpha1.AuthSpec{
+					Authenticators: &corev1alpha1.AuthenticatorsSpec{
+						SelfSigned: &corev1alpha1.SelfSignedSpec{
+							TokenLifetime: &metav1.Duration{
+								// Since we have a token expiration buffer of one minute,
+								// a new token will constantly be fetched.
+								Duration: time.Second * 1,
+							},
+						},
+					},
+				},
 			},
 		},
 	}
-
+	// Initial run
+	setSecuritySpecDefaults(cluster)
 	err := driver.PreInstall(cluster)
 	require.NoError(t, err)
 
 	// Default configuration should be loaded with autogenerated secret keys. ConfigMap
 	securitySecret := &v1.Secret{}
 	err = testutil.Get(k8sClient, securitySecret, pxutil.SecurityPXAuthKeysSecretName, cluster.Namespace)
 	require.NoError(t, err)
+	require.Equal(t, 64, len(securitySecret.Data[component.SecuritySystemSecretKey]))
+	require.Equal(t, 64, len(securitySecret.Data[component.SecuritySharedSecretKey]))
+	oldSystemSecret := securitySecret.Data[component.SecuritySystemSecretKey]
+	oldSharedSecret := securitySecret.Data[component.SecuritySharedSecretKey]
+	require.NotEqual(t, securitySecret.Data[component.SecuritySystemSecretKey], securitySecret.Data[component.SecuritySharedSecretKey])
 
-	require.Equal(t, 64, len(securitySecret.StringData[component.SecuritySystemSecretKey]))
-	require.Equal(t, 64, len(securitySecret.StringData[component.SecuritySharedSecretKey]))
-	require.NotEqual(t, securitySecret.StringData[component.SecuritySystemSecretKey], securitySecret.StringData[component.SecuritySharedSecretKey])
+	// No changes should happen to the auto-generated secrets
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+	err = testutil.Get(k8sClient, securitySecret, pxutil.SecurityPXAuthKeysSecretName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Equal(t, oldSystemSecret, securitySecret.Data[component.SecuritySystemSecretKey])
+	require.Equal(t, oldSharedSecret, securitySecret.Data[component.SecuritySharedSecretKey])
+
+	// Token secrets should be created and be valid jwt tokens
+	adminSecret := &v1.Secret{}
+	err = testutil.Get(k8sClient, adminSecret, component.SecurityPXAdminTokenSecretName, cluster.Namespace)
+	require.NoError(t, err)
+	_, _, err = new(jwt.Parser).ParseUnverified(string(adminSecret.Data[component.SecurityAuthTokenKey]), &jwt.StandardClaims{})
+	require.NoError(t, err)
+
+	userSecret := &v1.Secret{}
+	err = testutil.Get(k8sClient, userSecret, component.SecurityPXUserTokenSecretName, cluster.Namespace)
+	require.NoError(t, err)
+	_, _, err = new(jwt.Parser).ParseUnverified(string(userSecret.Data[component.SecurityAuthTokenKey]), &jwt.StandardClaims{})
```

Comment:
Should we check the contents of the token to ensure we are setting the right values?
Things like expiry, role, groups, etc?